sound/soc/sof/ipc3-topology.c

```diff
@@ -1594,6 +1595,17 @@ static int sof_ipc3_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 	if (ret < 0)
 		goto free;
 
+	/* Subtract the base to match the IPC3 FW dai index. */
+	if (comp_dai->type == SOF_DAI_INTEL_ALH) {
+		comp_dai->dai_index -= desc->sdw_bidir_base;
+		if (comp_dai->dai_index < 0) {
```

Comment:
dai_index is uint32_t ...

Reply:
> dai_index is uint32_t ...

Thanks, move the check before subtracting.